packages/blocks/src/api/raw-handling/iframe-remover.js

```diff
@@ -15,3 +15,17 @@ export default function( node ) {
 		remove( node );
 	}
 }
+
+export function checkForUntransformedIframe( rawTransform, node ) {
+	const iframeRegex = /(?:<iframe[^>]*)(?:(?:\/>)|(?:>.*?<\/iframe>))/gi;
+
+	return (
+		( ! rawTransform &&
+			( node.nodeName === 'IFRAME' ||
+				iframeRegex.test( node.innerHTML ) ) ) ||
+		( rawTransform &&
+			node.nodeName === 'FIGURE' &&
+			rawTransform.blockName === 'core/html' &&
+			iframeRegex.test( node.innerHTML ) )
+	);
+}
```

Comment:
I think I will split this check into a couple of sub methods as a bit hard to work out at a glance what is happening here